### ISSUE TITLE
BUG: MSK IAM auth not working

### ISSUE THREAD

**Issue by JohnPreston** [conduktor/conduktor-platform#105]
Hello

I have deployed the platform with a very simple MSK cluster, which uses SASL IAM authentication.
I am getting nothing to work and no IAM error that indicates that IAM errors (permissions denied) would be the source of the problem.

Here is some logs with ERROR. It seems it is not finding the class for this.

Docker image tag is `conduktor/conduktor-platform@sha256:d9a147958b5e6f5a00cf6c88421f3741b5aa66ce179ecf2ddc456631ff6e7368`

```
testinggg/svc/ecs/testinggg/conduktorplatform conduktor-platform/conduktor-platform/c2b8695bed634a4faf6cb2d2c10d0427 2022-11-16 14:22:09,235 ERROR c.l.k.ConsumerGroupCollector$ akka://kafka-lag-exporter/user/consumer-group-collector-amazon-msk-iam - Supervisor RestartSupervisor saw failure [28]: Invalid value io.conduktor.aws.IAMClientCallbackHandler for configuration sasl.client.callback.handler.class: Class io.conduktor.aws.IAMClientCallbackHandler could not be found. org.apache.kafka.common.config.ConfigException: Invalid value io.conduktor.aws.IAMClientCallbackHandler for configuration sasl.client.callback.handler.class: Class io.conduktor.aws.IAMClientCallbackHandler could not be found.
```

Configuration file (minus the interpolated values). Yes I checked the value for bootstrap is correct.

```yaml
files:
  /etc/conduktor/config.yaml:
    context: jinja2
    content: |
      auth:
        local-users:
          - email: noone@you.me
            password: ReplaceM3!!
            groups:
              - ADMIN
      organization:
        name: ${ORG_NAME:-testing}
      clusters:
        - id: amazon-msk-iam
          name: Amazon MSK IAM
          color: #FF9900
          bootstrapServers: {{ msk_bootstrap(env_var('MSK_CLUSTER_ARN'), 'BootstrapBrokerStringSaslIam') }}
          properties: |
            security.protocol=SASL_SSL
            sasl.mechanism=AWS_MSK_IAM
            sasl.jaas.config=software.amazon.msk.auth.iam.IAMLoginModule required;
            sasl.client.callback.handler.class=io.conduktor.aws.IAMClientCallbackHandler
```

The platform is running with an ECS Task so it has a "native" IAM role to perform all the API calls.

Way to replicate:

Create `conduktor.yaml` from 

```yaml
version: "3.8"

x-msk_cluster:
  new-cluster:
    Properties:
      ClusterName: testingcluster
      KafkaVersion: "3.3.1"
      NumberOfBrokerNodes: 2
      BrokerNodeGroupInfo:
        StorageInfo:
          EBSStorageInfo:
            VolumeSize: 10
        InstanceType: "kafka.m5.large"
        ClientSubnets:
          - x-vpc::vpc::StorageSubnets
      ClientAuthentication:
        Unauthenticated:
          Enabled: False
        Sasl:
          Iam:
            Enabled: true
          Scram:
            Enabled: false
        Tls:
          Enabled: false
      EncryptionInfo:
        EncryptionInTransit:
          InCluster: true
          ClientBroker: TLS
    MacroParameters:
      StorageScaling:
        MaxInGB: 2048
        Target: 42.0
    Services:
      conduktor-platform:
        Access:
          MSKCluster: RO
        KafkaAccess:
          Iam:
            topic:
              Admin:
                - "*"
            group:
              Admin:
                - "*"
            transactional-id:
              Producer:
                - "*"

services:
  conduktor-platform:
    image: conduktor/conduktor-platform:${CDK_VERSION:-latest}
    ports:
      - 8080:8080
    volumes:
      - conduktor_data:/var/conduktor
      - conduktor_config:/etc/conduktor
    networks:
      public:
    x-network:
      AssignPublicIp: true
    environment:
      CDK_IN_CONF_FILE: /etc/conduktor/config.yaml
      CDK_LISTENING_PORT: 8080
      PLATFORM_LISTENING_PORT: 8080
      RUN_MODE: small
      CDK_DEBUG: "false"
    deploy:
      replicas: 1
      labels:
        ecs.task.family: conduktor-platform
      resources:
        reservations:
          cpus: 2.0
          memory: 4GB
        limits:
          memory: 16GB
    healthcheck:
      test: ["CMD-SHELL", "curl -sq --fail http://localhost:8080/platform/api/modules/health/live"]
      interval: 30s
      start_period: 120s # Leave time for the psql init scripts to run
      timeout: 5s
      retries: 3
    x-ecs:
      CapacityProviderStrategy:
        - CapacityProvider: FARGATE
          Base: 1
          Weight: 1
    depends_on:
      - conduktor-platform-config
    x-scaling:
      Range: 1-2

  conduktor-platform-config:
    image: public.ecr.aws/compose-x/ecs-files-composer
    deploy:
      labels:
        ecs.task.family: conduktor-platform
        ecs.depends.condition: SUCCESS
    volumes:
      - conduktor_config:/etc/conduktor
    environment:
      MSK_CLUSTER_ARN: x-msk_cluster::new-cluster::ClusterArn
      ECS_CONFIG_CONTENT: |
        files:
          /etc/conduktor/config.yaml:
            context: jinja2
            content: |
              auth:
                local-users:
                  - email: noone@you.me
                    password: ReplaceM3!!
                    groups:
                      - ADMIN
              organization:
                name: ${ORG_NAME:-testing}
              clusters:
                - id: amazon-msk-iam
                  name: Amazon MSK IAM
                  color: #FF9900
                  bootstrapServers: {{ msk_bootstrap(env_var('MSK_CLUSTER_ARN'), 'BootstrapBrokerStringSaslIam') }}
                  properties: |
                    security.protocol=SASL_SSL
                    sasl.mechanism=AWS_MSK_IAM
                    sasl.jaas.config=software.amazon.msk.auth.iam.IAMLoginModule required;
                    sasl.client.callback.handler.class=io.conduktor.aws.IAMClientCallbackHandler

volumes:
  conduktor_config: {}
  conduktor_data: {}

networks:
  public:
    x-vpc: PublicSubnets

x-vpc:
  Properties:
    DisableNat: True

```

Install & deploy

```shell
python3 -m venv venv
source venv/bin/activate
pip install pip -U
pip install git+https://https://github.com/compose-x/ecs_composex_msk_cluster@0.23-module-sample
pip install git+https://github.com/compose-x/ecs_composex@refactor/resources-and-stacks
ecs-compose-x init
ecs-compose-x up -d templates -f conduktor.yaml -p conduktor-iam-testing
```

PS: to connect to the platform, go to ECS -> clusters -> conduktor-iam-testing -> tasks -> $(find the right task) -> Public IP address.

In the security group, add ingress from your own IP to 8080/tcp

EDIT: changed the consumer group permissions to reflect latest tests done on my end.

**Comment by JohnPreston** [comment]
Hmm, I just noticed the `consumer-group` to be set to `my-consumer-group-app`. Let me correct that to `*` though.

**Comment by qboileau** [comment]
I think that you should use `software.amazon.msk.auth.iam.IAMClientCallbackHandler`  instead of `io.conduktor.aws.IAMClientCallbackHandler` in `sasl.client.callback.handler.class` property. 

You can check test project https://github.com/conduktor/quickstart-conduktor-aws-msk that should work with latest version of the platform.

**Comment by JohnPreston** [comment]
Solved by #106
